### PR TITLE
WIP: Deploy fixtures.pulproject.org on www

### PR DESCRIPTION
**Pull request by mscherer**
See https://pulp.plan.io/issues/6638

### CODE REVIEW

**Comment by bmbouter** [comment]
This PR looks resonable to me. I'm ok to merge it if you also think so. What needs to happen before this can be merged so we can test it from the fixtures.pulpproject.org site?

**Comment by mscherer** [comment]
Well, for a start, I would like to know what happen if that thing is down. For example, if we need to upgrade the host, and reboot it, would the whole CI go down ?

Then I do also want to make sure than others people can fix it. The setup is dealing with containers, and they tend to fill hard drive quite fast, and then systems behave in a weird way (like, yesterday, my dns/dhcp on openbsd couldn't run python because /var was full, so much fun to debug at 1h in the morning), so if the answer to the first question is "this is critical", we might want to spent time in a more redundant setup, and some training for folks who would help in case of problem.

Finally, we also need to verify it work, so if someone could run the test suite against the setup, to valide, it would help a lot.

**Comment by bmbouter** [comment]
The CI uses local fixtures not these. While it would be nice, I'm not concerned about high availability of this container currently.

We'd like to do some testing of it, but is it already deployed for us to test against?

**Comment by bmbouter** [comment]
Oh I see it is deployed: https://fixtures.pulpproject.org/fixtures/  This is great!

**Comment by bmbouter** [comment]
@mscherer we'll sanity test it now. Assuming that tests ok, let us know what else we should do to get this merged. Thank you again!

**Review by dkliban — APPROVED** [review]
I tested using https://fixtures.pulpproject.org/fixtures/ and my tests were able to download fixtures.

File: roles/fixtures/files/deploy_fixtures.sh
Comment (daviddavis) [inline roles/fixtures/files/deploy_fixtures.sh]: I would probably use docker.io/pulp/pulp-fixtures. We've seen some major outages with quay.io in the past few weeks.

File: roles/fixtures/files/fixtures.service
Comment (daviddavis) [inline roles/fixtures/files/fixtures.service]: Here too.

**Comment by mscherer** [comment]
Ok I guess I should not hardcode it. I will fix that

**Comment by mscherer** [comment]
So, i did deployed the new one, but it seems to have made the server full, so I stopped the vhost for now. I didn't had time to look at cleaning the image yet, but I am working on redploying with docker as a source.

**Comment by mscherer** [comment]
So one of the issue is that the doc take lots of space. I did ran fdupes, and there is a few duplicates. With the current setup, we will run out of a space quite soon, so I will increase the disk in the mean time.

**Comment by dkliban** [comment]
@mscherer We currently have a cron job that publishes the image daily. However, we don't actually need to do that and will stop doing so. We will only publish when changes are made. This should reduce the frequency of new layers being added. 

I also suspect that as new images are pulled down, we are creating orphaned layers on the host running the container. Perhaps we should run a cron job that cleans up orphaned image layers?

**Comment by daviddavis** [comment]
FWIW, I've already disabled the cron job that was publishing the image nightly. Now it should only publish a new image if any changes have been made.

**Comment by mscherer** [comment]
There is a cron job that do run podman prune, but yeah, I do need to see if that clean up what I think it should clean up (spoiler: "likely not").

**Comment by bmbouter** [comment]
@mscherer where does `{{image}}` come from?

**Comment by daviddavis** [comment]
@mscherer some of the fixture data contained hostname so we had to add a `BASE_URL` environment variable that can be set at runtime:

```
podman run --rm -p 8080:80 -e BASE_URL=https://fixtures.pulpproject.org/  --name fixtures "{{ image }}"
```

Any chance you could update this?

**Comment by mscherer** [comment]
Yup

**Comment by mscherer** [comment]
I also missed @bmbouter comment, good catch, it was a file that I forgot to include.

**Comment by mscherer** [comment]
So, for some reason, podman image prune wasn't run (or if it ran, it didn't work). I am going to verify what is going on. 

Also, service fixtures restart fail with a error message:
```
juin 17 05:27:13 www.pulpproject.org podman[29149]: Error: error creating container storage: the container name "fixtures" is already in use by "792e4df5dc52d19f65686eab56f1e1f7c95c2b8afe010834b2d11d0faaa3f87a".
```

but podman ps -a show nothing, kinda start to suspect a bug. So I wiped again /var/lib/containers.

**Comment by daviddavis** [comment]
@mscherer I messed up. It should be `BASE_URL=https://fixtures.pulpproject.org` and not `BASE_URL=https://fixtures.pulpproject.org/`.

**Comment by mscherer** [comment]
ok, going to fix

**Comment by bmbouter** [comment]
@mscherer and @daviddavis do you agree this ready for merging or is there more to be done?

**Comment by daviddavis** [comment]
:+1: to merging

**Comment by duck-rh** [comment]
So IIUC `BASE_URL` is sensitive to a trailing slash? That does not seem very safe. At the very least we should add a comment to avoid breaking things in the future.

**Comment by daviddavis** [comment]
I'll update our code to trim the trailing slash if it exists.

**ETA**: https://github.com/pulp/pulp-fixtures/pull/172

**Comment by bmbouter** [comment]
@duck-rh @mscherer what do you think about merging this?

**Review by duck-rh — APPROVED** [review]

**Pull request merged**
